node/node.go

```diff
@@ -212,7 +212,35 @@ type Node struct {
 // Assumes [n.CPUTracker] and [n.CPUTargeter] have been initialized.
 func (n *Node) initNetworking(primaryNetVdrs validators.Set) error {
 	currentIPPort := n.Config.IPPort.IPPort()
-	listener, err := net.Listen(constants.NetworkType, fmt.Sprintf(":%d", currentIPPort.Port))
+
+	// Default to binding to all interfaces on the specified port
+	listenAddress := fmt.Sprintf(":%d", currentIPPort.Port)
+
+	// If a listen host is provided, bind only to that address on the specified port.
+	//
+	// Providing either loopback address - `::1` for ipv6 and `127.0.0.1` for ipv4 - will avoid the
+	// need for a firewall exception on recent MacOS:
+	//
+	//   - MacOS requires a manually-approved firewall exception [1] for each version of a given
+	//   binary that wants to bind to all interfaces (i.e. with an address of `:[port]`). Each
+	//   compiled version of avalanchego requires a separate exception to be allowed to bind to all
+	//   interfaces.
+	//
+	//   - A firewall exception is not required to bind to a loopback interface, but the only way for
+	//   Listen() to bind to loopback for both ipv4 and ipv6 is to bind to all interfaces [2] which
+	//   requires an exception.
+	//
+	//   - Thus, the only way to start a local avalanche network on MacOS without approving one or
+	//   more firewall exceptions for the targeted avalanchego binaries is to bind to loopback for
+	//   only a single network type by specifying either `::1` (ipv6) or `127.0.0.1` (ipv4).
+	//
+	// 1: https://apple.stackexchange.com/questions/393715/do-you-want-the-application-main-to-accept-incoming-network-connections-pop
+	// 2: https://github.com/golang/go/issues/56998
+	if len(n.Config.ListenHost) != 0 {
+		listenAddress = net.JoinHostPort(n.Config.ListenHost, fmt.Sprintf("%d", currentIPPort.Port))
+	}
```

Comment:
2 small nits here:

1. `net.JoinHostPort` handles when the host is `""` exactly how our default does - so we can just ditch the check entirely.
2. I feel like it's a bit weird to talk about multiple node instances in this comment... (This is all inside the node package where we only manage a single node). I feel like the comment is just as good (if not more concise) if we just talk about the single node case.

```suggestion
	// If a listen host is provided, bind only to that address on the specified port.
	//
	// Providing either loopback address - `::1` for ipv6 and `127.0.0.1` for ipv4 - will avoid the
	// need for a firewall exception on recent MacOS:
	//
	//   - MacOS requires a manually-approved firewall exception [1] for a given binary that wants
	//   to bind to all interfaces (i.e. with an address of `:[port]`).
	//
	//   - A firewall exception is not required to bind to a loopback interface, but the only way for
	//   Listen() to bind to loopback for both ipv4 and ipv6 is to bind to all interfaces [2] which
	//   requires an exception.
	//
	//   - Thus, the only way to start a local avalanche network on MacOS without approving a firewall
	//   exception is to bind to loopback for only a single network type by specifying either
	//   `::1` (ipv6) or `127.0.0.1` (ipv4).
	//
	// 1: https://apple.stackexchange.com/questions/393715/do-you-want-the-application-main-to-accept-incoming-network-connections-pop
	// 2: https://github.com/golang/go/issues/56998
	listenAddress := net.JoinHostPort(n.Config.ListenHost, fmt.Sprintf("%d", currentIPPort.Port))
```